### PR TITLE
Add preview to companies and domains

### PR DESCRIPTION
**Pull request by defild9**

### CODE REVIEW

**Review by VanyaRPP — APPROVED** [review]

File: common/components/UI/RealEstateComponents/RealEstateModal/RealEstateForm/index.tsx [hunk 1]
```diff
@@ -41,94 +44,113 @@ const RealEstateForm: FC<Props> = ({ form, currentRealEstate }) => {
         label="Назва компанії"
         rules={validateField('required')}
       >
-        <Input placeholder="Опис" maxLength={256} className={s.formInput} />
+        {editable ? (
+          <Input placeholder="Опис" maxLength={256} className={s.formInput} />
+        ) : (
+          <div>{companyNameValue}</div>
+        )}
```
Comment (vakot) [inline common/components/UI/RealEstateComponents/RealEstateModal/RealEstateForm/index.tsx:51]: @myershov prefers to user ```<Input disabled={!editable}/>``` in this cases instead (in rest form items also)

File: common/components/UI/RealEstateComponents/RealEstateModal/RealEstateForm/index.tsx [hunk 2]
```diff
@@ -9,19 +9,22 @@ import { IExtendedRealestate } from '@common/api/realestateApi/realestate.api.ty
...
+  disabled?: boolean
 }
 
-const RealEstateForm: FC<Props> = ({ form, currentRealEstate }) => {
-  const initialValues = useInitialValues(currentRealEstate)
+const RealEstateForm: FC<Props> = ({
+  form,
+  currentRealEstate,
+  editable = true,
+  disabled = false,
```
Comment (vakot) [inline common/components/UI/RealEstateComponents/RealEstateModal/RealEstateForm/index.tsx:21]: Form `disabled` is not require in our cases. All situations now can be handled by `editable` only

**Review by vakot — CHANGES_REQUESTED** [review]

File: common/components/UI/RealEstateComponents/RealEstateModal/RealEstateForm/index.tsx [hunk 3]
```diff
@@ -9,19 +9,22 @@ import { IExtendedRealestate } from '@common/api/realestateApi/realestate.api.ty
...
   return (
-    <Form
-      form={form}
-      layout="vertical"
-      className={s.Form}
-      initialValues={initialValues}
-    >
+    <Form form={form} layout="vertical" className={s.Form}>
       {currentRealEstate ? (
         <Form.Item name="domain" label="Надавач послуг">
```
Comment (vakot) [inline common/components/UI/RealEstateComponents/RealEstateModal/RealEstateForm/index.tsx:29]: Something happened to DomainSelect placeholder. Check this in UI, please
image

**Review by vakot — CHANGES_REQUESTED** [review]

File: common/components/UI/Reusable/EmailSelect.tsx [hunk 4]
```diff
@@ -31,14 +33,14 @@ export default function EmailSelect({ form }: EmailSelectProps) {
     <Form.Item
       name="adminEmails"
       label="Адміністратори"
-      rules={[
+      rules={!disabled && [
```
[the remark this answers is not in the document]
Comment (defild9) [reply common/components/UI/Reusable/EmailSelect.tsx:36]: It's need for hide star which show is required input in preview mode:
image
If we will be in edit mode or create:
image

File: common/components/UI/RealEstateComponents/RealEstateModal/index.tsx [hunk 5]
```diff
@@ -97,6 +97,7 @@ const RealEstateModal: FC<Props> = ({
       onCancel={closeModal}
       okText={currentRealEstate ? 'Зберегти' : 'Додати'}
       cancelText={'Відміна'}
+      okButtonProps={{ style: { display: !editable ? 'none' : 'inline' } }}
```
Comment (vakot) [inline common/components/UI/RealEstateComponents/RealEstateModal/index.tsx:100]: To hide button better use `okText=''` (empy `string` will hide the button)

Comment (defild9) [reply common/components/UI/RealEstateComponents/RealEstateModal/index.tsx:100]: if I use the okText button with an empty string, then the close button shifts to the left
image
But if i use  okButtonProps={{ style: { display: !editable ? 'none' : 'inline' } }} button in needed position
image

Comment (vakot) [reply common/components/UI/RealEstateComponents/RealEstateModal/index.tsx:100]: Ok, agreed. Also some new info, the `okText=''` method isn't working on antd version 5.x, so...
But, let's make it look like this `{ ...(!editable && { display: 'none' }) }` to not override default antd button display style by using conditional object field

File: common/components/UI/DomainsComponents/DomainModal/index.tsx [hunk 6]
```diff
@@ -60,8 +74,10 @@ const DomainModal: FC<Props> = ({ currentDomain, closeModal }) => {
       onCancel={closeModal}
       okText={currentDomain ? 'Зберегти' : 'Додати'}
       cancelText={'Відміна'}
+      okButtonProps={{ style: { display: !editable ? 'none' : 'inline' } }}
```
Comment (vakot) [inline common/components/UI/DomainsComponents/DomainModal/index.tsx:77]: Change this to conditional field also

File: common/components/UI/DomainsComponents/DomainModal/DomainForm/DomainStreets/index.tsx [hunk 7]
```diff
@@ -3,29 +3,29 @@ import { validateField } from '@common/assets/features/validators'
...
+
+const DomainStreets: React.FC<DomainStreetsProps> = ({ disabled = false }) => {
   const { data: streets, isLoading } = useGetAllStreetsQuery({})
 
   return (
     <Form.Item
       name="streets"
       label="Закріплені адреси"
-      rules={validateField('required')}
+      rules={!disabled && validateField('required')}
```
Comment (vakot) [inline common/components/UI/DomainsComponents/DomainModal/DomainForm/DomainStreets/index.tsx:17]: This approach can cause a runtime error. Get rid of this, please. Leave `validateField('required')` only

File: common/components/UI/Reusable/EmailSelect.tsx (hunk 4 above)
Comment (vakot) [inline common/components/UI/Reusable/EmailSelect.tsx:36]: Remove `!disabled &&` condition from all rules fields all across your changed files, please

**Review by vakot — CHANGES_REQUESTED** [review]

**Review by vakot — APPROVED** [review]

**Pull request merged**